pennylane_lightning/src/bindings/Bindings.hpp

```diff
@@ -227,4 +229,57 @@ template <class PrecisionT, class ParamT, class PyClass>
 void registerKernelsToPyexport(PyClass &pyclass) {
     registerKernelsToPyexportIter<PrecisionT, ParamT, 0>(pyclass);
 }
+
+/**
+ * @brief Return basic information of the compiled binary.
+ */
+auto getCompileInfo() -> pybind11::dict {
+    using namespace Util::Constant;
+    using namespace pybind11::literals;
+
+    const std::string_view cpu_arch_str = [] {
+        switch (cpu_arch) {
+        case CPUArch::AMD64:
+            return "AMD64";
+        case CPUArch::PPC64:
+            return "PPC64";
+        case CPUArch::ARM:
+            return "ARM";
+        default:
+            return "Unknown";
+        }
+    }();
+
+    const std::string_view compiler_name_str = [] {
+        switch (compiler) {
+        case Compiler::GCC:
+            return "GCC";
```

Comment:
We may also want to update the enums and add NVHPC to this, and maybe even nvcc.

Reply:
I am gonna add them. Bug it seems like nvc++ does not fully support C++17 yet... (I just failed to compile to project with nvc++)

Reply:
We can chat about this later in the week, as this may be worth giving them feedback on.